app/src/test/js/ctrlTestUtility.js

```diff
@@ -0,0 +1,75 @@
+function loadTest(scope, response) {
+    expect(scope.entities || scope.searchKeys).toEqual(response.data.tree);
+    expect(scope.parent).toEqual(response.data.tree);
+}
+
+function crumbTest(scope, response) {
+    expect(scope.crumbs.length).toEqual(1);
+    expect(scope.crumbs[0]).toEqual(response.data.tree);
+    expect(scope.parent).toEqual(response.data.tree);
+
+    scope.gotoCrumb(scope.crumbs[0]);
```

Comment:
I feel like this should be a separate test, `it("should navigate to a crumb")`. Also, if the state before and after is the same, you aren't actually testing that `gotoCrumb` works... it could do nothing and this test would pass.